orttraining/orttraining/core/optimizer/megatron_transformer.h

```diff
@@ -15,13 +15,17 @@ class MegatronTransformer : public GraphTransformer {
                       std::unordered_map<std::string, std::string>& updated_weight_names,
                       std::unordered_set<std::string>& weights_to_train,
                       std::unordered_map<std::string, training::TrainingSession::PartitionInfo>& weight_partition_info,
+                      training::TrainingSession::OptimizerState& initial_optimizer_states,
+                      const IExecutionProvider& execution_provider, // Required to get allocator for optimizer partitioning                     
                       const std::unordered_set<std::string>& compatible_execution_providers = {}) noexcept
       : GraphTransformer("MegatronTransformer", compatible_execution_providers),
         horizontal_parallel_rank_(horizontal_parallel_rank),
         horizontal_parallel_size_(horizontal_parallel_size),
         updated_weight_names_(updated_weight_names),
         weights_to_train_(weights_to_train),
-        weight_partition_info_(weight_partition_info) {}
+        weight_partition_info_(weight_partition_info),
+        initial_optimizer_states_(initial_optimizer_states),
+        execution_provider_(execution_provider) {}
```

Comment:
not sure why we need the execution_provider parameters here.

Reply:
we need this to get the CPU allocator which is required to allocate the partitioned tensor for optimizer state for column partitioning